Mathlib/Data/UInt.lean

```diff
@@ -5,15 +5,15 @@ import Mathlib.Algebra.GroupWithZero.Defs
 import Mathlib.Algebra.Ring.Basic
 import Mathlib.Data.ZMod.Defs
 
-lemma UInt8.val_eq_of_lt {a : Nat} : a < UInt8.size -> (ofNat a).val = a := Nat.mod_eq_of_lt
+lemma UInt8.val_eq_coe {a : Nat} : (ofNat a).val = a := rfl
 
-lemma UInt16.val_eq_of_lt {a : Nat} : a < UInt16.size -> (ofNat a).val = a := Nat.mod_eq_of_lt
+lemma UInt16.val_eq_coe {a : Nat} : (ofNat a).val = a := rfl
 
-lemma UInt32.val_eq_of_lt {a : Nat} : a < UInt32.size -> (ofNat a).val = a := Nat.mod_eq_of_lt
+lemma UInt32.val_eq_coe {a : Nat} : (ofNat a).val = a := rfl
 
-lemma UInt64.val_eq_of_lt {a : Nat} : a < UInt64.size -> (ofNat a).val = a := Nat.mod_eq_of_lt
+lemma UInt64.val_eq_coe {a : Nat} : (ofNat a).val = a := rfl
 
-lemma USize.val_eq_of_lt {a : Nat} : a < USize.size -> (ofNat a).val = a := Nat.mod_eq_of_lt
+lemma USize.val_eq_coe {a : Nat} : (ofNat a).val = a := rfl
```

Comment:
Is there a reason you changed the statement rather than fixing the syntax here? What actually changed is that lean now puts the coercion on the LHS, but the lemma was hoping it would appear on the RHS.

Reply:
Oh, I missed that the statement had changed. I wish our default for lemmas that are *actually about coercions* was that we always wrote the coercion explicitly in the statement, to make it obvious what the lemma is actually saying.

I've fixed this now, and this PR becomes trivial.

Reply:
>  I wish our default for lemmas that are actually about coercions was that we always wrote the coercion explicitly in the statement, to make it obvious what the lemma is actually saying.

Unfortunately there are still cases where writing the coercion in the place where the pretty-printer shows it actually changes the meaning of the lemma! An obvious example is the coercion on morphisms, which in my experience fails to roundtrip at all every time.

Reply:
Okay, but that is not a reason to not write the coercions wherever possible!